src/java/org/apache/cassandra/index/sai/disk/v2/hnsw/CassandraOnDiskHnsw.java

```diff
@@ -52,47 +54,42 @@
 import org.apache.lucene.index.VectorEncoding;
 import org.apache.lucene.util.hnsw.HnswGraphSearcher;
 import org.apache.lucene.util.hnsw.NeighborQueue;
-import org.apache.lucene.util.hnsw.RandomAccessVectorValues;
 
 public class CassandraOnDiskHnsw extends JVectorLuceneOnDiskGraph
 {
     private static final Logger logger = LoggerFactory.getLogger(CassandraOnDiskHnsw.class);
+    private static final VectorTypeSupport vts = VectorizationProvider.getInstance().getVectorTypeSupport();
 
-    private final Supplier<VectorsWithCache> vectorsSupplier;
     private final OnDiskOrdinalsMap ordinalsMap;
     private final OnDiskHnswGraph hnsw;
     private final VectorSimilarityFunction similarityFunction;
-    private final VectorCache vectorCache;
 
     private static final int OFFSET_CACHE_MIN_BYTES = 100_000;
+    private final OnDiskVectors vectors;
     private FileHandle vectorsFile;
 
     public CassandraOnDiskHnsw(SegmentMetadata.ComponentMetadataMap componentMetadatas, PerIndexFiles indexFiles, IndexContext context) throws IOException
     {
         super(componentMetadatas, indexFiles);
+        logger.warn("Opening HNSW index -- we thought they were all upgraded to DiskANN");
```

Comment:
FWIW, we've had an alert running since early March to catch any HNSW indexes. It hasn't fired since then. We can probably remove this code soon.